### ISSUE TITLE
Player continues walking in one direction even when arrow key is released

### ISSUE THREAD

**Issue by nmadadair** [TwilioQuest/twilioquest-api-academy#13]
When walking the player into a collision (around edge of Catacombs and around trees outside in the Academy grounds map), at a certain point the player moves on their own even after the arrow key is released. Eventually this stops and goes back to normal. 

See screen record for more: This video shows the player walking downwards into the wall of the Catacombs when no arrow keys are currently being pressed. When I move the player to the left using the left arrow key, the player walks to the left until the collision is out of the when and then continues down. In the open doorway, I have to press the up arrow to force the player to go against the "stuck" down direction. Eventually the stuck direction stops.

https://user-images.githubusercontent.com/109177652/192376161-8db816b2-e46d-495a-8589-b986b3330419.mov

**Comment by MarvelProgramming** [comment]
Having trouble replicating this myself. Do you remember if anything else took place just before this bug occurred?

**Comment by nmadadair** [comment]
@MarvJohnson Yes, I was doing something super specific. I had been checking for missing collisions in the map (or incorrectly layered tiles that would allow the player to walk over or under something incorrectly). When I was checking for the collision layers around the edges of the map (the trees outside or the walls in the Catacombs), I walked persistently, by pressing the arrow keys over and over or holding them down in one direction, into the edges of the map. Eventually the player began to move on its own in that direction. That being said, I'm also struggling to reproduce in the same locations.

**Comment by nmadadair** [comment]
@kerrielake and I experienced this recently when testing in one of the API maps

**Comment by MarvelProgramming** [comment]
> @kerrielake and I experienced this recently when testing in one of the API maps

I've been experimenting with this as a bit, and the closest I've gotten to reproducing it is by holding the "command" key after letting go of any movement key (e.g. walking in any direction, then holding down the "command" key, then letting go of the "w", "a", "s", and/or "d" key(s)). It takes a lot of finagling, but with enough of that I can get the player character to continue moving in some direction without doing it myself. Still not entirely sure what the deal is there, though

**Comment by kerrielake** [comment]
It happened to me again today when I was still holding down the directional buttons while trying to press "Help" to open dev tools

**Comment by MarvelProgramming** [comment]
Looks like this can happen if you hold down a movement key, move focus outside of the game window while it's still accepting input (e.g. clicking on any of the window menu options, using command tab, etc), letting go of all movement keys, and giving focus back to the game window. I'll leave some video examples here for the record!


https://user-images.githubusercontent.com/51891409/208013204-3b649966-c7f8-476c-ab14-848850ea835d.mov


https://user-images.githubusercontent.com/51891409/208013369-f75122a4-1e95-4b6f-8156-4308767a62af.mov
